### ISSUE TITLE
docs(linter/jsx-a11y): missing docs for flattened object configs

### ISSUE THREAD

**Issue by baevm** [oxc-project/oxc#22533]
2 `jsx-a11y` rules are missing documentation for flattened object configuration:
1. https://oxc.rs/docs/guide/usage/linter/rules/jsx_a11y/no-interactive-element-to-noninteractive-role.html#configuration
https://github.com/oxc-project/oxc/blob/5fa47746a3201304efeda9ae4968c3e915fc73e0/crates/oxc_linter/src/rules/jsx_a11y/no_interactive_element_to_noninteractive_role.rs#L32-L38

2. https://oxc.rs/docs/guide/usage/linter/rules/jsx_a11y/no-noninteractive-element-interactions.html#configuration - This rule is also missing the default values for the handlers option in the generated docs.
https://github.com/oxc-project/oxc/blob/5fa47746a3201304efeda9ae4968c3e915fc73e0/crates/oxc_linter/src/rules/jsx_a11y/no_noninteractive_element_interactions.rs#L84-L92

**Comment by camc314** [comment]
partially fixed by https://github.com/oxc-project/schemars/pull/45

(docs aren't quite right yet, but the actual type is ok)
